libs/features/mas/web-components/src/variants/variant-layout.js

```diff
@@ -0,0 +1,103 @@
+import { html } from 'lit';
+
+export class VariantLayout {
+  static styleMap = {};
+
+  card;
+
+  insertVariantStyle() {
+    if (!VariantLayout.styleMap[this.card.variant]) {
+      VariantLayout.styleMap[this.card.variant] = true;
+      const styles = document.createElement('style');
+      styles.innerHTML = this.getGlobalCSS();
+      document.head.appendChild(styles);
+    }
+  }
+
+  constructor(card) {
+    this.card = card;
+    setTimeout(() => this.insertVariantStyle(), 1);
+  }
+
+  get badge() {
```

Comment:
Why this and the getters below do not stay in `merch-card.js` ? To me `VariantLayout` class looks not necessary.

Reply:
Because such methods affect layout. And layout is now deferred to the VariantLayout classes. Sometimes some are being overriden.